### PR TITLE
samples: benchmark: coremark: improve support for alternative configu…

### PR DESCRIPTION
**Pull request by kapi-no**
…rations

Improved support for alternative configurations in the CoreMark sample.

Ref: NCSDK-27522

### CODE REVIEW

File: samples/benchmarks/coremark/README.rst [hunk 1]
```diff
@@ -201,6 +201,67 @@ After flashing, messages describing the benchmark state will appear in the conso
...
+* Manual benchmark execution on a button press
+
+The following subsections describe alternative configurations.
+The deviation from the default configuration is described.
+Each subsection also provides a build command that demonstrates how to build the described alternative configuration.
+
+Flash and run configuration
+---------------------------
+
+The flash and run configuration changes the trigger for the benchmark execution. In this configuration, the benchmark is automatically started after the system boot.
```
Comment (zycz) [inline samples/benchmarks/coremark/README.rst:220]: It will start on all cores if I see it correctly.

`is automatically started  on all supported cores after the system boot.`

File: samples/benchmarks/coremark/README.rst [hunk 2]
```diff
@@ -201,6 +201,67 @@ After flashing, messages describing the benchmark state will appear in the conso
...
+
+The default sample configuration uses the following settings for the benchmark execution:
+
+* Memory allocation using the stack memory (see the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STACK` Kconfig option)
+* Single thread (see the :kconfig:option:`CONFIG_COREMARK_THREADS_NUMBER` Kconfig option)
+* Manual benchmark execution on a button press
+
+The following subsections describe alternative configurations.
+The deviation from the default configuration is described.
+Each subsection also provides a build command that demonstrates how to build the described alternative configuration.
```
Comment (zycz) [inline samples/benchmarks/coremark/README.rst:215]: Maybe lets add build command to be even more clear - we provide it for other configurations below :)

File: scripts/quarantine_integration.yaml [hunk 3]
```diff
@@ -521,13 +521,24 @@
   comment: "Configurations excluded to limit resources usage in integration builds"
 
 - scenarios:
-    - sample.benchmark.coremark_heap
+    - sample.benchmark.coremark.flash_and_run
   platforms:
+    - nrf52dk/nrf52832
```
Comment (zycz) [inline scripts/quarantine_integration.yaml:526]: Shouldn't we exclude here more platforms? I would leave just one as we check proper working in  sample.benchmark.coremark. Or we could leave all tests here and remove more platforms from  sample.benchmark.coremark

File: samples/benchmarks/coremark/sample.yaml [hunk 4]
```diff
@@ -61,8 +86,8 @@ tests:
       - nrf54l15dk/nrf54l15/cpuapp
       - nrf54h20dk/nrf54h20/cpuapp
     tags: ci_build sysbuild ci_samples_benchmarks
-    extra_args: EXTRA_CONF_FILE="prj_static_memory.conf"
-  sample.benchmark.coremark_multithread:
+    extra_args: FILE_SUFFIX=static_memory
+  sample.benchmark.coremark.multiple_threads:
```
Comment (zycz) [inline samples/benchmarks/coremark/sample.yaml:90]: Did you check correct working of those - I guess that there might be some problems with heap configurations on ppr core.

File: scripts/quarantine_integration.yaml (hunk 3 above)
Comment (zycz) [reply scripts/quarantine_integration.yaml:526]: I guess the second solution is better as we will have harness then.

File: samples/benchmarks/coremark/prj_flash_and_run.conf [hunk 5]
```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+# Common coremark configuration
+############################################################################
```
Comment (zycz) [inline samples/benchmarks/coremark/prj_flash_and_run.conf:7]: I guess there is no way to reuse common configuration :(
I don't like coping it but understand there is no better solution available if we want to use file sufix?

Comment (kapi-no) [reply samples/benchmarks/coremark/prj_flash_and_run.conf:7]: You can define a promptless `APP_COREMARK` Kconfig at the sample level that selects the common configuration and overrides the defaults for non-selectable options. However, I decided that this copy-paste approach is better as it cleanly demonstrates what needs to be enabled in the .conf file for the CoreMark sample.

File: samples/benchmarks/coremark/sample.yaml (hunk 4 above)
Comment (kapi-no) [reply samples/benchmarks/coremark/sample.yaml:90]: yes, they work. I don't understand why the heap configuration wouldn't work for PPR core.

Also, we are discussing heap configuration in the comment that was attached to the line with "multiple_threads" configuration

File: scripts/quarantine_integration.yaml (hunk 3 above)
Comment (kapi-no) [reply scripts/quarantine_integration.yaml:526]: the CI execution for the harness configuration hasn't changed

File: samples/benchmarks/coremark/README.rst (hunk 1 above)
Comment (kapi-no) [reply samples/benchmarks/coremark/README.rst:220]: yes, all alternative configurations are applied to the default image and all other non-default images (e.g. `coremark_cpuppr`, `coremark_cpurad`)

File: scripts/quarantine_integration.yaml (hunk 3 above)
Comment (kapi-no) [reply scripts/quarantine_integration.yaml:526]: Well, I think the default configuration should be tested at least as well as an alternative configuration ("flash_and_run"). Don't you think that?

Comment (kapi-no) [reply scripts/quarantine_integration.yaml:526]: Well, most users would probably work in 90% of cases with the default configuration so I would assume that this last configuration variant that should be broken.

The flash_and_run is a special case of runtime configuration that is also extensively tested as it is the only configuration that allows us to have runtime verification (the remaining test configurations are build-only).

TLDR: I would keep the increased integration test scope for the default configuration.

Comment (zycz) [reply scripts/quarantine_integration.yaml:526]: Ok, lets leave it then as it is :)

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 6]
```diff
@@ -412,6 +412,13 @@ Other samples
 
     * Configuration for the :ref:`zephyr:nrf54h20dk_nrf54h20` board to support multi-domain logging using the ARM Coresight STM.
     * The logging format in the standard logging mode to align it with the format used in the multi-domain logging mode.
+    * Support for alternative configurations to use the :ref:`file suffix feature from Zephyr <app_build_file_suffixes>`.
+      The following file suffixes are supported as alternative configurations:
+
+        * `flash_and_run`
+        * `heap_memory`
+        * `static_memory`
+        * `multiple_threads`
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:421]:
```suggestion
        * ``flash_and_run``
        * ``heap_memory``
        * ``static_memory``
        * ``multiple_threads``
```

File: samples/benchmarks/coremark/README.rst (hunk 1 above)
Comment (peknis) [inline samples/benchmarks/coremark/README.rst:220]:
```suggestion
The flash and run configuration changes the trigger for the benchmark execution.
In this configuration, the benchmark is automatically started after the system boot.
```

File: samples/benchmarks/coremark/README.rst [hunk 7]
```diff
@@ -201,6 +201,67 @@ After flashing, messages describing the benchmark state will appear in the conso
...
+
+.. parsed-literal::
+   :class: highlight
+
+   west build -b *board_target* -- -DFILE_SUFFIX=flash_and_run
+
+Heap memory configuration
+-------------------------
+
+The heap memory configuration changes the method for memory allocation during the benchmark execution. In this configuration, the benchmark allocates the memory using the heap memory (see the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` Kconfig option).
```
Comment (peknis) [inline samples/benchmarks/coremark/README.rst:232]:
```suggestion
The heap memory configuration changes the method for memory allocation during the benchmark execution.
In this configuration, the benchmark allocates the memory using the heap memory (see the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_MALLOC` Kconfig option).
```

File: samples/benchmarks/coremark/README.rst [hunk 8]
```diff
@@ -201,6 +201,67 @@ After flashing, messages describing the benchmark state will appear in the conso
...
+
+.. parsed-literal::
+   :class: highlight
+
+   west build -b *board_target* -- -DFILE_SUFFIX=heap_memory
+
+Static memory configuration
+---------------------------
+
+The static memory configuration changes the method for memory allocation during the benchmark execution. In this configuration, the benchmark allocates the memory using the static memory (see the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STATIC` Kconfig option).
```
Comment (peknis) [inline samples/benchmarks/coremark/README.rst:244]:
```suggestion
The static memory configuration changes the method for memory allocation during the benchmark execution.
In this configuration, the benchmark allocates the memory using the static memory (see the :kconfig:option:`CONFIG_COREMARK_MEMORY_METHOD_STATIC` Kconfig option).
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by alstrzebonski — APPROVED** [review]

File: samples/benchmarks/coremark/README.rst (hunk 1 above)
Comment (kapi-no) [reply samples/benchmarks/coremark/README.rst]: improved

File: samples/benchmarks/coremark/README.rst (hunk 2 above)
Comment (kapi-no) [reply samples/benchmarks/coremark/README.rst:222]: improved

**Review by zycz — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by miha-nordic — APPROVED** [review]
